crates/router/src/core/payments/operations/payment_start.rs

```diff
@@ -242,14 +242,24 @@ where
     #[instrument(skip_all)]
     async fn make_pm_data<'a>(
         &'a self,
-        _state: &'a AppState,
-        _payment_data: &mut PaymentData<F>,
+        state: &'a AppState,
+        payment_data: &mut PaymentData<F>,
         _storage_scheme: storage_enums::MerchantStorageScheme,
     ) -> RouterResult<(
         BoxedOperation<'a, F, api::PaymentsStartRequest>,
         Option<api::PaymentMethodData>,
     )> {
-        Ok((Box::new(self), None))
+        if payment_data
+            .payment_attempt
+            .connector
+            .clone()
+            .map(|connector_name| connector_name == *"bluesnap".to_string())
```

Comment:
It would be preferable to avoid connector specific implementation details leaking to the core payments flow code.

Reply:
If this check is not added, then it would fetch the payment method data for all other connectors in this flow which is not necessary.